portal-api/src/utils/string.js

```diff
@@ -12,7 +14,48 @@ const hasValue = (str) => {
   return false;
 };
 
+/**
+ * Objective:
+  The objective of the isValidEmail function is to validate whether a
+  given email address is valid or not. It checks if the input is a non-empty
+  string and then uses the joi library to validate the email format.
+
+Inputs:
+  email: a string representing the email address to be validated
+
+Flow:
+  1. Check if the email is a non-empty string
+  2. Use the joi library to validate the email format
+  3. Return true if the email is valid, false otherwise
+
+Outputs:
+  true: if the email is valid
+  false: if the email is invalid
+
+Additional aspects:
+  The function uses the joi library for email validation
+  The email must be at least 3 characters long and in a valid email format
+  The function returns false if the input is not a non-empty string
+
+ * @param {String} email Email address as string
+ * @returns Boolean
+ */
+const isValidEmail = (email) => {
+  if (typeof email === 'string' && email.length > 0) {
```

Comment:
`joi.required()` should handle the length, no?
and `joi.email()` should handle the string as well? i.e. whether it has the right format?

Reply:
Simplified the compound condition.